Dalamud/Game/Gui/ChatGui2.cs

```diff
@@ -0,0 +1,580 @@
+        this.PrintChat(
+            new XivChatEntry2
+            {
+                Message = message,
+                Type = this.configuration.GeneralChatType,
+            },
+            pluginInterface);
+    }
+
+    /// <summary>
+    /// Queue an error chat message. While method is named as Print, it only adds an entry to
+    /// the queue, later to be processed when UpdateQueue() is called.
+    /// </summary>
+    /// <param name="message">A message to send.</param>
+    /// <param name="pluginInterface">Your plugin's <see cref="DalamudPluginInterface"/>.</param>
+    public void PrintError(string message, DalamudPluginInterface pluginInterface)
+    {
+        // Log.Verbose("[CHATGUI PRINT REGULAR ERROR]{0}", message);
+        this.PrintChat(
+            new XivChatEntry2
+            {
+                Message = message,
+                Type = XivChatType2.Urgent,
+            },
+            pluginInterface);
+    }
+
+    /// <summary>
+    /// Queue an error chat message. While method is named as Print, it only adds an entry to
+    /// the queue, later to be processed when UpdateQueue() is called.
+    /// </summary>
+    /// <param name="message">A message to send.</param>
+    /// <param name="pluginInterface">Your plugin's <see cref="DalamudPluginInterface"/>.</param>
+    public void PrintError(SeString message, DalamudPluginInterface pluginInterface)
+    {
+        // Log.Verbose("[CHATGUI PRINT SESTRING ERROR]{0}", message.TextValue);
+        this.PrintChat(
+            new XivChatEntry2
+            {
+                Message = message,
+                Type = XivChatType2.Urgent,
+            },
+            pluginInterface);
+    }
+
+    /// <summary>
+    /// Process a chat queue.
+    /// </summary>
+    public void UpdateQueue()
+    {
+        while (this.chatQueue.Count > 0)
+        {
+            var chat = this.chatQueue.Dequeue();
+
+            if (this.baseAddress == IntPtr.Zero)
+            {
+                continue;
+            }
+
+            var senderRaw = (chat.Name ?? string.Empty).Encode();
+            using var senderOwned = this.libcFunction.NewString(senderRaw);
+
+            var messageRaw = (chat.Message ?? string.Empty).Encode();
+            using var messageOwned = this.libcFunction.NewString(messageRaw);
+
+            this.HandlePrintMessage(this.baseAddress, chat.Type, senderOwned.Address, messageOwned.Address, chat.Timestamp, chat.Parameters, chat.MessageSource, chat.SourceName, chat.MessagePrintedCallback);
+        }
+    }
+
+    /// <summary>
+    /// Queue a chat message. While method is named as Print, it only adds an entry to the queue,
+    /// later to be processed when UpdateQueue() is called.  This is the internal version that should be used by all Dalamud calls to properly set the message source.
+    /// </summary>
+    /// <param name="message">A message to send.</param>
+    internal void Print_Internal(string message)
```

Comment:
Would probably be happier with `PrintRaw` as a name instead of this.

Reply:
As long as it gets the point across that these, and only these, are the versions that should be called internally by Dalamud.  I guess depending on `FindCallingPlugin`s behavior when it's just Dalamud calling, it would eliminate the need for the internal versions period.